src/main/java/org/zalando/riptide/AsyncRest.java

```diff
@@ -59,17 +67,38 @@ public AsyncDispatcher execute(final HttpMethod method, final URI url, final Htt
     }
 
     private <T> AsyncDispatcher execute(final HttpMethod method, final URI url, final HttpEntity<T> entity) {
-        final List<HttpMessageConverter<?>> converters = template.getMessageConverters();
-        final Callback<T> callback = new Callback<>(converters, entity);
+        try {
+            final AsyncClientHttpRequest request = asyncClientHttpRequestFactory.createAsyncRequest(url, method);
+            RequestUtil.writeRequestEntity(entity, new AsyncClientHttpRequestAdapter(request), converters);
+            final ListenableFuture<ClientHttpResponse> responseFuture = request.executeAsync();
+            final ExceptionWrappingFuture future = new ExceptionWrappingFuture(responseFuture);
+
+            return new AsyncDispatcher(converters, future, router);
+        } catch (IOException ex) {
+            final String message = String.format("I/O error on %s request for \"%s\":%s", method.name(), url, ex.getMessage());
+            throw new ResourceAccessException(message, ex);
+        }
+    }
+
+    static class ExceptionWrappingFuture extends ListenableFutureAdapter<ClientHttpResponse, ClientHttpResponse> {
```

Comment:
This is the weird async error handling that @lukasniemeier mentioned. Needs a better name but works.